content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md

```diff
@@ -150,86 +154,87 @@ The upgrade workflow at high level is the following:
+    [upgrade/version] You have chosen to change the cluster version to "v1.18.0"
+    [upgrade/versions] Cluster version: v1.17.3
+    [upgrade/versions] kubeadm version: v1.18.0
     [upgrade/confirm] Are you sure you want to proceed with the upgrade? [y/N]: y
     [upgrade/prepull] Will prepull images for components [kube-apiserver kube-controller-manager kube-scheduler etcd]
     [upgrade/prepull] Prepulling image for component etcd.
     [upgrade/prepull] Prepulling image for component kube-apiserver.
     [upgrade/prepull] Prepulling image for component kube-controller-manager.
     [upgrade/prepull] Prepulling image for component kube-scheduler.
-    [apiclient] Found 0 Pods for label selector k8s-app=upgrade-prepull-kube-scheduler
-    [apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-apiserver
     [apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-controller-manager
     [apiclient] Found 0 Pods for label selector k8s-app=upgrade-prepull-etcd
-    [apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-scheduler
+    [apiclient] Found 0 Pods for label selector k8s-app=upgrade-prepull-kube-scheduler
+    [apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-apiserver
     [apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-etcd
+    [apiclient] Found 1 Pods for label selector k8s-app=upgrade-prepull-kube-scheduler
     [upgrade/prepull] Prepulled image for component etcd.
-    [upgrade/prepull] Prepulled image for component kube-controller-manager.
     [upgrade/prepull] Prepulled image for component kube-apiserver.
+    [upgrade/prepull] Prepulled image for component kube-controller-manager.
     [upgrade/prepull] Prepulled image for component kube-scheduler.
     [upgrade/prepull] Successfully prepulled the images for all the control plane components
-    [upgrade/apply] Upgrading your Static Pod-hosted control plane to version "v1.17.0"...
-    Static pod: kube-apiserver-luboitvbox hash: 8d931c2296a38951e95684cbcbe3b923
-    Static pod: kube-controller-manager-luboitvbox hash: 2480bf6982ad2103c05f6764e20f2787
-    Static pod: kube-scheduler-luboitvbox hash: 9b290132363a92652555896288ca3f88
+    [upgrade/apply] Upgrading your Static Pod-hosted control plane to version "v1.18.0"...
+    Static pod: kube-apiserver-myhost hash: 2cc222e1a577b40a8c2832320db54b46
+    Static pod: kube-controller-manager-myhost hash: f7ce4bc35cb6e646161578ac69910f18
+    Static pod: kube-scheduler-myhost hash: e3025acd90e7465e66fa19c71b916366
     [upgrade/etcd] Upgrading to TLS for etcd
-    [upgrade/staticpods] Writing new Static Pod manifests to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests446257614"
+    [upgrade/etcd] Non fatal issue encountered during upgrade: the desired etcd version for this Kubernetes version "v1.18.0" is "3.4.3-0", but the current etcd version is "3.4.3". Won't downgrade etcd, instead just continue
```

Comment:
Is this expected in the output? Or is this a bug in the way the semver is being compared somewhere?

Reply:
yes, it's a known minor bug, when the version of the client is the same as the request target upgrade, yet one of them has a revision `-0`.
https://github.com/kubernetes/kubernetes/pull/88845